src/frontend/Parse.ml

```diff
@@ -208,6 +221,8 @@ let%expect_test "parse indices, two different colons" =
   let ast =
     parse_string Parser.Incremental.program
       "model { matrix[5, 5] x; print(x[2 - 3 ? 3 : 4 : 2]); }"
+    |> Result.map_error ~f:render_syntax_error
+    |> Result.ok_or_failwith
```

Comment:
Does it make sense to make these handlers already part of the parsing function? I just keep seeing this combination repeated.

Reply:
Yes, the idea is to hide of this from stanc by using the `Frontend.S` module signature. I can't do that right now (reason outlined above) so tried to keep things as-is until we make the change.